### ISSUE TITLE
rabbitmq_prometheus: Add authentication support

### ISSUE THREAD

**Issue by SimonUnge** [rabbitmq/rabbitmq-server#8605]
### Is your feature request related to a problem? Please describe.

Some prometheus monitoring systems supports target authentication, such as the big one (https://prometheus.io/docs/prometheus/latest/configuration/configuration/) that supports basic_auth and oauth.

I suggest we enable opt-in support in `rabbit_prometheus_handler` to authenticate with `rabbit_mgmt_util:is_authorized_monitor`.

adding a `prometheus.authentication.enabled` flag to its cuddlefish schema for example?

I have working code ready for review if this issue gets accepted.

### Describe the solution you'd like

Call `rabbit_mgmt_util:is_authorized_monitor` in `is_authorized` if authentication is enabled, otherwise return true.

### Describe alternatives you've considered

Do nothing.

**Comment by michaelklishin** [comment]
No objections from me.

**Comment by tomaThomas** [comment]
This is completed by #8661, isn't it?

**Comment by mkuratczyk** [comment]
Yes. Seems like it's not documented - I will add it to the docs shortly.
You can set `prometheus.authentication.enabled = true` in the config fule
